cmd/demo/node.go

```diff
@@ -258,14 +262,57 @@ func (n *node) listen() error {
 	if err != nil {
 		return errors.WithMessage(err, "could not start tcp listener")
 	}
-	go n.client.HandleChannelProposals(n)
+	db, err := leveldb.LoadDatabase(config.DatabasePath)
+	if err != nil {
+		return errors.WithMessage(err, "creating/loading database")
+	}
+	persister, err := keyvalue.NewPersistRestorer(db)
+	if err != nil {
+		return errors.WithMessage(err, "creating PersistRestorer")
+	}
+	n.client.EnablePersistence(persister)
+
+	go n.client.Handle(n, n)
+	n.client.OnNewChannel(func(ch *client.Channel) {
+		n.log.WithField("ID", ch.ID()).Trace("OnNewChannel")
+		go ch.Watch()
+		// If so, update the channel of the peer, when a peer
+		// is associated to this channel.
+		for _, p := range n.peers {
+			if p.ch != nil && p.ch.ID() == ch.ID() {
+				p.ch = newPaymentChannel(ch, func() {
+					n.handleFinal(p)
+				})
```

Comment:
You're producing code duplication logic of how a `newPaymentChannel` is setup here (see below). Please avoid logic duplication.